libs/dh/wholesale/feature-search/src/lib/form/dh-wholesale-form.component.ts

```diff
@@ -62,7 +62,7 @@ export class DhWholesaleFormComponent implements OnInit, OnDestroy {
       {
         start: sub(new Date().setHours(0, 0, 0, 0), { days: 10 }).toISOString(),
         end: zonedTimeToUtc(
-          new Date().setHours(0, 0, 0, 0),
+          new Date().setHours(23, 59, 59, 999),
```

Comment:
Question: is ‘setHours’ still necessary?

Reply:
Not really but it might be useful for mentally mapping the initial value with the value written back to the model.

It doesn't matter what value is set to "end", it will automatically be converted to end of day on initialization.

Say it was `new Date()` (`2022-11-02T13:05:37.038Z`) before initializing the component, afterward initialization the model will change to `2022-11-02T22:59:59.999Z`. Or something like `2022-11-01T23:00:00Z` came from the API, it will also be adjusted to `2022-11-02T22:59:59.999Z`.

So while technically it's not necessary, it might be desired for consistency.

If we decide to keep it, we can extract it to a util method (say `setToEndOfDay`) so developers don't have to use `setHours` directly.

What do you think?

Reply:
To be consistent with how the datepicker in Material works, the initial value is not changed. So whatever is set to the model, that will be used in the datepicker. Changing the date afterwards (either by selecting it from the UI or manually typing it) will result in the "to" date always being set to end of the day.

Reply:
After the latest changes, the "end" date is always set to "end of selected day" internally in the datepicker component on initialization. So it doesn't matter what value is set to `end` initially in the component where the datepicker is used.